docs/source/hooks/common_use_cases.md

```diff
@@ -113,3 +113,70 @@ def after_dataset_loaded(self, dataset_name: str, data: Any) -> None:
     end = time.time()
     self._logger.info("Loading dataset %s ended at %0.3f", dataset_name, end)
 ```
+
+## Use Hooks to load external credentials
+We recommend using the `after_context_created` Hook to add credentials to the session's config loader instance. In this example we show how to load credentials from Azure KeyVault.
```

Comment:
Link to KeyVault?

Reply:
Is it worth saying something like other stores are supported (examples) and explain what differences there may be (if any)?